File: year2018/day01/python/solution.py
def main():
    print(part_1())
    print(part_2())


def part_1():
    return sum(input)

def part_2():
    frequencies, frequency = set(), 0

    while True:
        for number in input:
            frequency += number
            
            if frequency in frequencies:
                return frequency

            frequencies.add(frequency)


def setup(path):
    global input
    
    with open(path, 'r') as file:
        input = [int(num) for num in file]


if __name__ == '__main__':
    setup('../puzzle.txt')
    main()

File: year2018/day01/java/Solution.java
import java.nio.file.*;
import java.util.*;

public class Solution {
  private static int[] NUMBERS;

  public static void main(String[] args) {
    try {
      setup(Files.readString(Paths.get("../puzzle.txt")));
      System.out.println(part1());
      System.out.println(part2());
    } catch(Exception e) {
      e.printStackTrace();
    }
  }

	public static int part1() {
    int sum = 0;
    for (int num : NUMBERS) {
      sum += num;
    }
    return sum;
  }

  public static int part2() {
    Set<Integer> frequencies = new HashSet<>();
    int frequency = 0;
    while (true) {
      for (int num : NUMBERS) {
        if (frequencies.contains(frequency += num)) {
          return frequency;
        }
        frequencies.add(frequency);
      }
    }
  }

  public static void setup(String input) {
    NUMBERS = Arrays.stream(input.split("\n")).mapToInt(Integer::parseInt).toArray();
  }
}